presto-main/src/main/java/io/prestosql/operator/aggregation/BucketReadFunction.java

```diff
@@ -0,0 +1,20 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.operator.aggregation;
+
+@FunctionalInterface
+public interface BucketReadFunction<K>
```

Comment:
Let's rename this to `BucketConsumer` and the method below to `processBucket`, or simply `process`. Calling it a "function" is misleading, as this doesn't return any value.